common/app/common/dfp/HighMerchandiseComponentAgent.scala

```diff
@@ -7,8 +7,8 @@ trait HighMerchandiseComponentAgent {
 
   protected def targetedHighMerchandisingLineItems: Seq[HighMerchandisingLineItem]
 
-  def hasHighMerchAdAndTag(adUnitSuffix:String, tags: Seq[Tag]) = {
+  def hasHighMerchAdAndTagandEdition(adUnitSuffix:String, tags: Seq[Tag],edition:String) = {
```

Comment:
Now that we are checking three things, enumerating each check in the method name becomes a bit untenable; maybe consider renaming the method to something like `isTargetedByHighMerch` and then ensure that the subsequent code is clear as to what that means i.e. a combination of `matchesAdUnit`, `matchesTag` and `matchesEdition` etc